controller_manager/src/controller_manager_parameters.yaml

```diff
@@ -65,6 +70,11 @@ controller_manager:
             }
       controllers:
         periodicity:
+          window_size: {
+            type: int,
+            default_value: -1,
+            description: "The number of samples to consider for the mean. The diagnostics will be computed over the last `window_size` samples. -1 means that all samples will be considered.",
+          }
           mean_error:
```

Comment:
```
controllers:
  window_size:
  periodicity:
    mean_error:
```
For the controllers, I believe this window_size should be a level higher, as it is being used for both periodicity and execution_time